fe/fe-core/src/main/java/com/starrocks/connector/elasticsearch/ElasticsearchConnector.java

```diff
@@ -43,11 +43,15 @@ public ElasticsearchConnector(ConnectorContext contex) {
     @Override
     public ConnectorMetadata getMetadata() {
         if (metadata == null) {
-            try {
```

Comment:
thanks for your patch. Have you ever had a race-related issue in here?

Reply:
I haven't encounter this kind of problem until now in our production, I found it when I review related codes, and I think make it more robust may be better.